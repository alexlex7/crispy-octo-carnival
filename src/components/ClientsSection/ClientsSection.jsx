import React from 'react';
import ClientList from '../ClientList/ClientList';
import Container from '../Container/Container';
import PropTypes from 'prop-types';
import styles from './ClientsSection.module.css';
import { roundNumber } from '../../helpers/roundNumber';

export default function ClientsSection({ items, isMobile, totalClients }) {
  return (
    <Container>
      <div className={styles.wrapper}>
        <div className={styles.titleWrapper}>
          <h2 className={styles.title}>
            Trusted by {roundNumber(totalClients)}+ companies
          </h2>
        </div>
        <div className={styles.listWrapper}>
          <ClientList items={items} isMobile={isMobile} />
        </div>
      </div>
    </Container>
  );
}

ClientsSection.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      imageURL: PropTypes.string.isRequired,
      mobileImageURL: PropTypes.string.isRequired,
    })
  ),
  isMobile: PropTypes.bool,
  totalClients: PropTypes.string.isRequired,
};
