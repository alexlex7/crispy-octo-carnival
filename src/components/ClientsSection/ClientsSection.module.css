.listWrapper {
  display: flex;
  justify-content: center;
  align-items: center;
}

.title {
  margin: 0;

  font-family: 'DM Sans';
  font-style: normal;
  font-weight: 700;
  color: #244d4d;
}

@media screen and (max-width: 1204px) {
  .wrapper {
    padding: 60px 0px 98px 0px;
    display: flex;
    flex-direction: column;
    align-items: center;
    row-gap: 24px;
  }
  .title {
    font-size: 18px;
    line-height: 23px;
  }
  .titleWrapper {
    margin-bottom: 24px;
  }
}

@media screen and (min-width: 1205px) {
  .wrapper {
    padding: 140px 0px;
    display: flex;
    flex-direction: column;
    align-items: center;
    row-gap: 40px;
  }
  .title {
    font-size: 32px;
    line-height: 42px;
  }
  .titleWrapper {
    margin-bottom: 40px;
  }
}
