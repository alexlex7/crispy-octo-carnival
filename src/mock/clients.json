[
  {
    "id": "1",
    "name": "Mastercard",
    "imageURL": "http://localhost:8080/images/clients/masterCard.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/masterCardMobile.jpg"
  },
  {
    "id": "2",
    "name": "Airbnb",
    "imageURL": "http://localhost:8080/images/clients/airbnb.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/airbnbMobile.jpg"
  },
  {
    "id": "3",
    "name": "Uber",
    "imageURL": "http://localhost:8080/images/clients/uber.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/uberMobile.jpg"
  },
  {
    "id": "4",
    "name": "PayPal",
    "imageURL": "http://localhost:8080/images/clients/paypal.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/paypalMobile.jpg"
  },
  {
    "id": "5",
    "name": "Visa",
    "imageURL": "http://localhost:8080/images/clients/visa.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/visaMobile.jpg"
  },
  {
    "id": "6",
    "name": "Stripe",
    "imageURL": "http://localhost:8080/images/clients/stripe.jpg",
    "mobileImageURL": "http://localhost:8080/images/clients/visaMobile.jpg"
  }
]
