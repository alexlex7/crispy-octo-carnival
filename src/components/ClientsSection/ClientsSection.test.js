import { render, screen } from '@testing-library/react';
import ClientsSection from './ClientsSection';
import clients from '../../mock/clients.json';

test('render clients section component', () => {
  const { asFragment } = render(
    <ClientsSection items={clients} isMobile={false} totalClients={'24'} />
  );
  const linkElement = screen.getByRole('list');
  expect(linkElement).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
