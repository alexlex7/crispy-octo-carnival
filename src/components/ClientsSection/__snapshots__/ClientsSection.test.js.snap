// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render clients section component 1`] = `
<DocumentFragment>
  <div
    class="container "
  >
    <div
      class="wrapper"
    >
      <h2
        class="title"
      >
        Trusted by 20+ companies
      </h2>
      <div
        class="listWrapper"
      >
        <ul
          class="desktopList"
        >
          <li
            class="listItem"
          >
            <img
              alt="Mastercard"
              src="http://localhost:8080/images/clients/masterCard.jpg"
              title="Mastercard"
            />
          </li>
          <li
            class="listItem"
          >
            <img
              alt="Airbnb"
              src="http://localhost:8080/images/clients/airbnb.jpg"
              title="Airbnb"
            />
          </li>
          <li
            class="listItem"
          >
            <img
              alt="Uber"
              src="http://localhost:8080/images/clients/uber.jpg"
              title="Uber"
            />
          </li>
          <li
            class="listItem"
          >
            <img
              alt="PayPal"
              src="http://localhost:8080/images/clients/paypal.jpg"
              title="PayPal"
            />
          </li>
          <li
            class="listItem"
          >
            <img
              alt="Visa"
              src="http://localhost:8080/images/clients/visa.jpg"
              title="Visa"
            />
          </li>
          <li
            class="listItem"
          >
            <img
              alt="Stripe"
              src="http://localhost:8080/images/clients/stripe.jpg"
              title="Stripe"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
